### ISSUE TITLE
Semantics-changing refactoring suggested by Sourcery "Use previously assigned local variable"

### ISSUE THREAD

**Issue by bwagner** [sourcery-ai/sourcery#208]
### Issue description or question

```Python
#!/usr/bin/env python
import pytest
import sys


def do_it(start, finish):
    cur = start
    while cur != finish:
        """
        Semantics-changing refactoring suggested by Sourcery:
        "Use previously assigned local variable":
        Replaces variable 'start' below by variable 'cur'.
        """
        if cur != start:
            return start
        cur += 1


def test1():
    assert 0 == do_it(0, 2)


if __name__ == "__main__":
    pytest.main([sys.argv[0]])

```
### Code editor or IDE name and version

Pycharm v 2021.3.2 (Professional Edition) Build #PY-213.6777.50, built on January 27, 2022

### OS name and version

OSX 11.6 (Big Sur)

**Comment by reka** [comment]
Hey @bwagner ,

Thanks a lot for reporting this bug :bug:  and the great example.
We've reproduced the bug and are working on a fix.
We'll let you know when it gets released.

**Comment by reka** [comment]
We've implemented a fix for this. It will be released in the version 0.11.2 in the coming weeks.
